### DISCUSSION TITLE
NSIS bootstrapper with conditional prerequisites

### DISCUSSION THREAD

**Discussion by adrian0b** [oleg-shilo/wixsharp#1672] [Q&A]
Hi,

I want to create an installer that bundles some driver installation files together with my application.
Basically I do something like this:

```
var setup = new NsisBootstrapper()
{
  Prerequisite =
  {
    FileName = @".\\res\CP210xWindows\USBXpressInstaller_x86.exe"
  },
  PrimaryFile = productMsi,
  OutputFile = $"whatever.exe"
};
```
My challenge now is that i need to provide an x86 and an x64 version of the driver installer.
I could simple build two installers, one for each but I am trying to figure out a way to bundle both driver installers and install one of them accoring to the target machines system.

Is there a way to do that?

Thank you,

Adriano

**Comment by geghamyan** [comment]
Hi Adriano,
Pack driver installers as two separate payloads, as in this example here [payloads](https://github.com/oleg-shilo/wixsharp/blob/69bf82e496b715b34f64dd19ecf7d875776e63d7/Source/src/WixSharp.Samples/Wix%23%20Samples/Bootstrapper/NsisBootstrapper/setup.cs#L177l) and write a tiny app (that's going to your prerequisite) that will run one of them based on the architecture.

Hope this helps,
Tigran

**Comment by adrian0b** [comment]
Hi Tigran,

thank you for your input. I will certainly try that.

How can I access the prerequesite Payloads from my prerequesite app? It seems this isn't shown in the example.
Is it just placed in the same folder?

Thanks,
Adriano

**Comment by geghamyan** [comment]
I don't have a dev setup ready to confirm, but I am fairly certain it will be in the same folder as the prerequisite executable, extracted to a temporary folder within the windows main temp directory.
